src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -64233,6 +64235,9 @@ static void M()
 
             var comp = CreateCompilation(new[] { source }, options: WithNonNullTypesTrue());
             comp.VerifyDiagnostics(
+                // (5,31): warning CS8600: Converting null literal or possible null value to non-nullable type.
+                //     static void F1(string x = (string)null!, string y = ((string)null)!) { }
+                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "(string)null!").WithLocation(5, 31),
```

Comment:
A) Does this solve the above comment?
B) I'm not sure I agree with this comment in general: there's a `!` here. Seems like this warning should be surpressed.

Reply:
Correct, this warning is wrong. We need another scenario which assigns `(string)null`.

Reply:
To properly solve suppression cases, warn on bad conversions, and so on, the best thing will be to just fix parameter default values more generally in the next release.

Right now, when we determine a parameter's "real" default value, we go through a destructive round-tripping where we bind the default value on the parameter declaration, then convert to ConstantValue, then convert back to BoundExpression.